packages/fast-components/src/form-associated/index.ts

```diff
@@ -0,0 +1,263 @@
+import { attr, FastElement } from "@microsoft/fast-element";
+import { keyCodeEnter } from "@microsoft/fast-web-utilities";
+
+/**
+ * Disable member ordering to keep property callbacks
+ * grouped with property declaration
+ */
+/* tslint:disable:member-ordering */
+export abstract class FormAssociated<
+    T extends HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement
+> extends FastElement {
+    /**
+     * Must evaluate to true to enable elementInternals.
+     * Feature detects API support and resolve respectively
+     */
+    public static get formAssociated(): boolean {

```

Comment:
We should perform this check once and cache the result so it can be used across all form elements and all properties that branch based on this condition.

Reply:
added a cached property - let me know if there was a more efficient way to do that though.